File: graphene-reader/src/test/net/iponweb/disthene/reader/utils/WildcardUtilTest.java
package net.iponweb.disthene.reader.utils;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class WildcardUtilTest {

  @Test
  public void shouldReturnEncodedMetricPath() {
    assertThat(WildcardUtil.getPathsRegExFromWildcard("hosts.service.role"), is("hosts\\.service\\.role"));
    assertThat(WildcardUtil.getPathsRegExFromWildcard("hosts.service.role.*"), is("hosts\\.service\\.role\\.[^\\.]*"));
  }
}

File: graphene-common/src/test/kotlin/net/iponweb/disthene/reader/utils/WildcardUtilTest.kt
package net.iponweb.disthene.reader.utils

import org.junit.Test

import org.hamcrest.CoreMatchers.`is`
import org.junit.Assert.assertThat

class WildcardUtilTest {

  @Test
  fun shouldReturnEncodedMetricPath() {
    assertThat(WildcardUtil.getPathsRegExFromWildcard("hosts.service.role"), `is`("hosts\\.service\\.role"))
    assertThat(WildcardUtil.getPathsRegExFromWildcard("hosts.service.role.*"), `is`("hosts\\.service\\.role\\.[^\\.]*"))
  }
}
